Add loading reactive (#3509)
Textualize/textual: docs/guide/widgets.md

    ```{.textual path="docs/examples/guide/widgets/tooltip02.py" hover="Button"}
    ```

## Loading indicator

> Widgets have a [`loading`][textual.widget.Widget.loading] reactive which when set to `True` will temporarily replace your widget with a [LoadingIndicator](../widgets/loading_indicator.md).

Reviewer: ```suggestion
Widgets have a [`loading`][textual.widget.Widget.loading] reactive which when set to `True` will temporarily replace your widget with a [`LoadingIndicator`](../widgets/loading_indicator.md).
```

You can use this to indicate to the user that the app is currently working on getting data, and there will be content when that data is available.
Let's look at an example of this.

=== "loading01.py"

    ```python title="loading01.py"
    --8<-- "docs/examples/guide/widgets/loading01.py"
    ```

    1. Shows the loading indicator in place of the data table.
    2. Insert a random sleep to simulate a network request.
    3. Show the new data.

=== "Output"

    ```{.textual path="docs/examples/guide/widgets/loading01.py"}
    ```

> In this example we have four [DataTable](../widgets/data_table.md) widgets, which we put in to a loading state by setting the widget's `loading` property to `True`.

Reviewer: ```suggestion
In this example we have four [`DataTable`](../widgets/data_table.md) widgets, which we put into a loading state by setting the widget's `loading` reactive to `True`.
```

Author: It's a reactive property. The term `reactive` could be confused with the descriptor object.

This will temporarily replace the widget with a loading indicator animation.

> When the (simulated) data has been retrieved, we reset the `loading` property to show the new data.

Reviewer: ```suggestion
When the (simulated) data has been retrieved, we reset the `loading` reactive to show the new data.
```
